### PR TITLE
Added Go 1.18.10 to Github Workflow and removed Godep dependencies.

### PR DESCRIPTION
**Pull request by klugorosado**
### Changes
- Added Go 1.18.10 to Github Workflow and removed Godep dependencies.
- Changed Makefile to build with Go Modules instead of govvv.
- Removed Godep dependencies.
- ran go mod tidy -> go mod vendor

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -1,8 +1,6 @@
 # github.com/Azure/azure-docker-extension v0.0.0-20160802215703-0dd2f199467d
 ## explicit
 github.com/Azure/azure-docker-extension/pkg/vmextension
```
Comment (frank-pang-msft) [inline vendor/modules.txt:3]: were these removed using go mod commands?

File: Makefile [hunk 2]
```diff
@@ -21,14 +21,14 @@ binary: clean
 	fi
 	# Set CGO_ENABLED=0 for static binaries, note that another approach might be needed if dependencies change
 	# (see https://github.com/golang/go/issues/26492 for using an external linker if CGO is required)
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 govvv build -v \
+	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -v -mod=readonly \
 	  -ldflags "-X main.Version=`grep -E -m 1 -o '<Version>(.*)</Version>' misc/manifest.xml | awk -F">" '{print $$2}' | awk -F"<" '{print $$1}'`" \
 	  -o $(BINDIR)/$(BIN) ./main
 	cp ./misc/applicationhealth-shim ./$(BINDIR)
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 govvv build -v \
+	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -v -mod=readonly \
```
[the remark this answers is not in the document]
Comment (klugorosado) [reply Makefile:28]: `go build` replaced` govvv build` for newer versions of go. `-mod=readonly `will allow us to build the code base without modifying go.mod file when run.

**Review by frank-pang-msft — APPROVED** [review]
We should get second reviewer too. If Ravi doesn't have time (since OOF) I think Hafthor can take a look.

**Comment by frank-pang-msft** [comment]
Also, we need to keep v1 up to date with v2.

**Review by Hafthor — APPROVED** [review]

**Pull request merged**